janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java

```diff
@@ -146,9 +149,7 @@ public void run() {
                     SliceResult qr = queue.poll(TIME_PER_TRY,TimeUnit.MILLISECONDS); //Try very short time to see if we are done
                     if (qr==null) {
                         if (pullThreads[i].isFinished()) continue; //No more data to be expected
-                        int retryCount = 0;
-                        while (!pullThreads[i].isFinished() && retryCount < TIMEOUT_MS / TIME_PER_TRY && qr == null) {
-                            retryCount ++;
+                        while (!pullThreads[i].isFinished() && qr == null) {
```

Comment:
Didn't the `retryCount` ensure that this doesn't result in an infinite loop? You said in the issue that this fails sometimes for vertices with very many edges, but doesn't that mean now that the `TIMEOUT_MS` are ignored?

Reply:
Hi @FlorianHockmann, in my case, code was reaching to max retryCount before indexing is complete. I tried increasing the TIMEOUT_MS, but it failed too for vertexes with large number of edges. Eg: a vertex having ~500K edges, CQL KeyIterator was taking a while to return the next element, and this loop was bailing before then. Because we moved the `finished=true` in DataPuller into finally block, this should never run into an infinite loop, at the same time giving enough time for while loop to continue looking for SliceResult until DataPuller says it's finished.

Reply:
I tried to look into why this retry count was introduced and found the PR #1237 which fixed #1157.
@porunov Since you created #1157, could you take a look at this and check whether removing the timeout here could be problematic? Do you maybe have an easy way to ensure that it doesn't have negative effects on the issue you described in #1157?
Otherwise I'm OK with the explanation @gguttikonda gave and would accept this PR.

Reply:
@porunov Do you like to comment on this?

Reply:
Sorry I didn't get back on this earlier. I will review #2066 closer to weekends because this PR is obsolete now

Reply:
Replied here https://github.com/JanusGraph/janusgraph/pull/2066#pullrequestreview-391922488
This change doesn't reintroduce the issue. LGTM.